### ISSUE TITLE
SVT: Import option "Codewind liberty" can be confusing for the users

### ISSUE THREAD

**Issue by sujeilyfonseca** [eclipse/codewind#2665]
**Codewind version:** 0.9.0 & 0.11.0
**OS:** Any OS

**IDE extension version:** 0.9.0 & 0.11.0
**IDE version:** Code 1.44.0

**Description:** Import option "Codewind liberty" can be confusing for the users

If you select that option for a Default Open Liberty project, it will be imported as a WebSphere Liberty MicroProfile project. 

<img width="597" alt="Screen Shot 2020-04-13 at 5 33 31 PM" src="https://user-images.githubusercontent.com/49531960/79164310-ce786680-7dae-11ea-87fd-bf1200881dd9.png">

<img width="310" alt="Screen Shot 2020-04-13 at 5 33 55 PM" src="https://user-images.githubusercontent.com/49531960/79164319-d33d1a80-7dae-11ea-865a-c971ee12d579.png">

We should use Other > Java, but if there is an option named "Codewind liberty", users can wrongly select it for a Default Open Liberty project.

<img width="306" alt="Screen Shot 2020-04-13 at 5 47 40 PM" src="https://user-images.githubusercontent.com/49531960/79164356-e4862700-7dae-11ea-8e0c-73087b983984.png">

@jagraj

**Comment by tetchel** [comment]
yes, I think that option should read 'codewind microprofile'?

**Comment by tetchel** [comment]
That string comes from [this route](https://github.com/eclipse/codewind/blob/master/src/pfe/portal/routes/projectTypes.route.js#L82) which gets `liberty` from [the template index file](https://github.com/codewind-resources/codewind-templates/blob/master/devfiles/index.json#L7)

**Comment by stalleyj** [comment]
Totally agree that this is confusing. I think a full description would give the user the information required to select the right type. The project-types api was significantly changed for this issue https://github.com/eclipse/codewind/issues/292 to give full information for appsody stacks but not Codewind style. Resolving this will require modification to the API and also changes to the VSCode plugin I believe. @sujeilyfonseca have you looked at the behaviour in Eclipse or IntelliJ to if the experience is similar ?

**Comment by sujeilyfonseca** [comment]
@stalleyj, I verified the import feature using Eclipse (Linux) and IntelliJ (MacOS), and the user experience is different.

**Eclipse:** The IDE automatically prompted me to approve the type `Other (Codewind Basic Container)` and the `Java` language. 

![Screen Shot 2020-04-15 at 12 18 37 PM](https://user-images.githubusercontent.com/49531960/79362081-24b2ea00-7f14-11ea-8690-a8219d2b26ce.png)

Also, if the user clicks next to select another type and language, there is no `Codewind liberty` option. Instead, there is a `MicroProfile / Java EE` option, which is more accurate. 

![Screen Shot 2020-04-15 at 12 19 04 PM](https://user-images.githubusercontent.com/49531960/79362095-28467100-7f14-11ea-9af1-673f770975e3.png)


**IntelliJ:** The user experience is similar to the Eclipse Plugin.

<img width="473" alt="Screen Shot 2020-04-15 at 12 29 44 PM" src="https://user-images.githubusercontent.com/49531960/79362740-f550ad00-7f14-11ea-9f97-712dc9c9a8a0.png">

<img width="471" alt="Screen Shot 2020-04-15 at 12 30 05 PM" src="https://user-images.githubusercontent.com/49531960/79362745-f71a7080-7f14-11ea-8df7-3020d2c5347f.png">

**Comment by stalleyj** [comment]
It seems that we must be providing all the required information if Eclipse and IntelliJ can provide the desired user experience.  @tetchel Are you able to use the same info that the other IDEs use ?

**Comment by tetchel** [comment]
Yes. I meant to revisit this but forgot about the issue because it wasn't tagged with my area

**Comment by tetchel** [comment]
ok, I investigated this, and I still think the backend should be improved.

The reason the other two IDEs have better labels is that they have hard-coded mappings of the Codewind project types to user-friendly project types [(see here)](https://github.com/eclipse/codewind-eclipse/blob/master/dev/org.eclipse.codewind.core/src/org/eclipse/codewind/core/internal/constants/ProjectType.java#L25). VS Code doesn't have this.

A proper, scalable solution would be to provide a `label` for each project type, combined with the `description`. The Appsody stacks use both of these, so no frontend hard-coding is necessary.
![image](https://user-images.githubusercontent.com/8888438/80773649-83897d80-8b28-11ea-99ea-5d6e9180c988.png)

I think the Codewind project type should provide more info, the way the appsody ones do. The Codewind 'subtypes' are confusing because they are actually languages, which I can't recall the reason for (@makandre ?). But I think the following would serve our purpose, and wouldn't require redesigning the API:

Old:
```
    "projectType": "liberty",
    "projectSubtypes": {
      "items": [
        {
          "id": "java",
          "label": "java"
        }
      ]
    }
  }
```
New: 
```
    "projectType": "liberty",
    "projectSubtypes": {
      "label": "Codewind template",
      "items": [
        {
          "id": "java",
          "label": "Microprofile / Java EE"
          "version": "It would be great if our templates had versions",
          "description": "It would be great if our templates had descriptions"
        }
      ]
    }
  }
```

**Comment by makandre** [comment]
@tetchel It was to align all 3 styles (Codewind/Appsody/ODO) such that:

- first choice: select project type, i.e. `nodejs`, or `docker`, or `Appsody`, or `ODO`, etc
   (remember that for Codewind types, each "technology" is itself a project type)

- second choice:
   - if you picked `Appsody`, choices are the stacks, i.e. `nodejs-express` or `java-openliberty`
   - if you picked `ODO`, choices are the component types
   - if you picked a Codewind type, choices are the languages (tho this is only applicable for the `docker` type; in all other cases the language is implied)

**Note** the menu in VSCode is flatten out to a single list, this was due to feedback.

**Comment by tobespc** [comment]
ths information is here https://github.com/codewind-resources/codewind-templates/blob/master/devfiles/index.json

**Comment by stalleyj** [comment]
It is clear that style, type, subtype, language, framework are all being used and in slightly different ways for the different "styles". We need design/user experience input here so that we get this right going forward.

**Comment by tetchel** [comment]
I think that putting the data we already have (in the index.json Toby linked) into this API's response would solve this just fine

**Comment by thisguy-1** [comment]
please someone assign this to me

**Comment by tobespc** [comment]
@tetchel I think for now, with this being no where near the experience we want that until we get the UX sorted out, we hold off making any changes to pfe.
